Content.Server/VendingMachines/VendingMachineComponent.cs

```diff
@@ -9,16 +9,18 @@
 using Robust.Shared.ViewVariables;
 using Content.Server.VendingMachines.systems;
 using static Content.Shared.Wires.SharedWiresComponent;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 
 namespace Content.Server.VendingMachines
 {
     [RegisterComponent]
     public sealed class VendingMachineComponent : SharedVendingMachineComponent, IWires
     {
         public bool Ejecting;
+        public bool Emagged = false;
         public TimeSpan AnimationDuration = TimeSpan.Zero;
-        [DataField("pack")]
-        public string PackPrototypeId = string.Empty;
+        [ViewVariables] [DataField("pack", customTypeSerializer:typeof(PrototypeIdSerializer<VendingMachineInventoryPrototype>))]   public string PackPrototypeId = string.Empty;
+        [ViewVariables] [DataField("emagPack", customTypeSerializer:typeof(PrototypeIdSerializer<VendingMachineInventoryPrototype>))]   public string EmagPackPrototypeId = string.Empty;
         public string SpriteName = "";
```

Comment:
This seems incompatible with any form of machine upgrading and is better off being a part of emag to apply a new pack. Same with Emagged should be a separate component so it actually supports components being added at any time.

Reply:
How is this incompatible? What's stopping upgraded vendors from adding entries using the same method?